SmartIntersections/Patch/LoadPathsPatch.cs

```diff
@@ -0,0 +1,206 @@
+using System.Collections.Generic;
+using System.Reflection.Emit;
+using UnityEngine;
+
+namespace SmartIntersections.Patches
+{
+    //public static void LoadPaths(BuildingInfo info, ushort buildingID, ref Building data, float elevation)
+    [HarmonyPatch(typeof(BuildingDecoration), nameof(BuildingDecoration.LoadPaths))]
+    public class LoadPathsPatch
+    {
+        internal static WrappersDictionary _networkDictionary;
+        internal static ActionGroup _actionGroup;
+        internal static HashSet<ConnectionPoint> borderNodes;
+
+        // Called before intersection is built
+        internal static void ReleaseCollidingSegments()
+        {
+            // We obtain a list of nodes adjacent to the deleted segment to know where to reconnect
+            HashSet<ConnectionPoint> borderNodes = new HashSet<ConnectionPoint>();
+            if (ToolControllerDetour.CollidingSegmentsCache2 == null)
+                return;
+            foreach (ushort segment in ToolControllerDetour.CollidingSegmentsCache2)
+            {
+                try
+                {
+                    //Debug.Log("Releasing segment " + segment);
+                    NetSegment netSegment = NetUtil.Segment(segment);
+
+                    // We keep untouchable segments
+                    if ((netSegment.m_flags & NetSegment.Flags.Untouchable) != NetSegment.Flags.None)
+                        continue;
+
+                    bool inverted = ((netSegment.m_flags & NetSegment.Flags.Invert) != NetSegment.Flags.None);
+
+                    borderNodes.Add(new ConnectionPoint(netSegment.m_startNode, netSegment.m_startDirection, netSegment.Info, inverted));
+                    borderNodes.Add(new ConnectionPoint(netSegment.m_endNode, netSegment.m_endDirection, netSegment.Info, !inverted));
+
+                    WrappedSegment segmentW = _networkDictionary.RegisterSegment(segment);
+                    _actionGroup.Actions.Add(segmentW);
+                    segmentW.IsBuildAction = false;
+                    segmentW.Release();
+
+                    if (segmentW.StartNode.TryRelease())
+                    {
+                        _actionGroup.Actions.Add(segmentW.StartNode);
+                        segmentW.StartNode.IsBuildAction = false;
+                    }
+
+                    if (segmentW.EndNode.TryRelease())
+                    {
+                        _actionGroup.Actions.Add(segmentW.EndNode);
+                        segmentW.EndNode.IsBuildAction = false;
+                    }
+
+                    //NetUtil.ReleaseSegment(segment, true);
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError(e);
+                }
+            }
+
+            borderNodes.RemoveWhere(n => !NetUtil.ExistsNode(n.Node));
+
+            ToolControllerDetour.CollidingSegmentsCache2 = null;
+
+            //Debug.Log("Border nodes (1): " + borderNodes.Count);
+
+            borderNodes = borderNodes;
```

Comment:
There is the problem
(I've already fixed it)

Reply:
Oh god! wow! so it was the scope! I should learn my lesson so as to avoid such a mistake in future.